src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -145,6 +157,11 @@ struct OperationViewModel {
         self.requiredQueryParams = requiredQueryParams
         self.requiredHeaders = requiredHeaders
         self.uriParams = uriParams
+        self.clientMethodOptionsViewModel = ClientMethodOptionsViewModel(
+            from: operation,
+            with: model,
+            properties: optionalParams
```

Comment:
So this should just pass in the optionalParams as `[Parameter]` and let `ClientMethodOptionsViewModel` create the view models from that. If at any point the ClientMethodOptionsViewModel might want more info about a parameter, it would likely not be in a ParameterViewModel. That's why it's best to just pass it all of the info and only strip the information down at the very end.

Reply:
Changed per feedback.